apps/backend/src/services/requestServices.ts

```diff
@@ -0,0 +1,199 @@
+import { Request } from '@prisma/client';
+import { CreateRequestData, UpdateRequestData } from '../types';
+import prismaClient from '../../prismaClient';
+import middleware from '../utils/middleware';
+
+/**
+ * performs narrowing on object `obj`
+ * returns `true` if `obj` contains required properties for
+ * `CreateRequestData` type
+ * @param obj - request.body
+ * @returns - type predicate (boolean)
+ */
+const isCreateRequestData = (obj: object): obj is CreateRequestData => (
+  'title' in obj && typeof obj.title === 'string'
+  && 'content' in obj && typeof obj.content === 'string'
+);
+
+/**
+ * - parses data sent to POST /requests
+ * - must contain neighborhood_id, title and content
+ * @param body req.body
+ * @returns Promise which resolves to parsed create request data
+ */
+const parseCreateRequestData = async (body: unknown): Promise<CreateRequestData> => {
+  if (!middleware.isObject(body)) {
+    const error = new Error('unable to parse data');
+    error.name = 'InvalidInputError';
+    throw error;
+  }
+
+  if (isCreateRequestData(body)) {
+    const requestData: CreateRequestData = {
+      title: body.title,
+      content: body.content,
+    };
+
+    return requestData;
+  }
+
+  const error = new Error('title or content missing or invalid');
+  error.name = 'InvalidInputError';
+  throw error;
+};
+
+/**
+ * - fetches request from db
+ * - throws error if request not found
+ * @param requestId
+ * @returns
+ */
+const getRequestById = async (requestId: number): Promise<Request> => {
+  const request: Request = await prismaClient.request.findUniqueOrThrow({
+    where: {
+      id: requestId,
+    },
+  });
+
+  return request;
+};
+
+const parseCloseRequestData = async (body: unknown): Promise<number> => {
+  if (!middleware.isObject(body)) {
+    const error = new Error('unable to parse data');
+    error.name = 'InvalidInputError';
+    throw error;
+  }
+
+  if ('neighborhood_id' in body
```

Comment:
Any special reason why we are passing neighborhood_id here via body and not URL. Do we even need neighborhood_id while closing the request? I am thinking that once user clicks on `ACCEPT` button in the response, the frontend will send a `PUT (or PATCH) /requests/:id/close`  and `neighborhoodId` will not be required.

I would leave it like this for now. If we do not need neighborhood_id in the future for closing the request, then we can modify this.

Reply:
It doesn't make sense to me either. As I said for the create request route, I think it makes more sense to obtain it through the URL if necessary. I'm not sure at this point whether it is necessary, though.

Reply:
Done.